src/pybind/mgr/rook/module.py

```diff
@@ -457,7 +457,17 @@ def _list_daemons(self,
         for p in pods:
             sd = orchestrator.DaemonDescription()
             sd.hostname = p['hostname']
-            sd.daemon_type = p['labels']['app'].replace('rook-ceph-', '')
+
+            # In Rook environments, the 'ceph-exporter' daemon is named 'exporter' whereas
+            # in the orchestrator interface, it is named 'ceph-exporter'. The purpose of the
+            # following adjustment is to ensure that the 'daemon_type' is correctly set.
+            # Without this adjustment, the 'service_to_daemon_types' lookup would fail, as
+            # it would be searching for a non-existent entry called 'exporter
+            if p['labels']['app'] == 'rook-ceph-exporter':
+                sd.daemon_type = 'ceph-exporter'
+            else:
+                sd.daemon_type = p['labels']['app'].replace('rook-ceph-', '')
```

Comment:
Should lines 467 and 469 be swapped? I thought we wanted to replace the label if the condition on line 466 is true.

Reply:
@travisn the order is correct as in fact what we are generating here is the `daemon_type`  which is normally the last part of the app name: i.e `mon` in case of `rock-ceph-mon` except for the `ceph-exporter` which has to be treated as special case.

Reply:
Ok I see now thanks for the explanation